server/sonar-process-monitor/src/main/java/org/sonar/application/SchedulerImpl.java

```diff
@@ -223,8 +222,13 @@ public void onAppStateOperational(ProcessId processId) {
 
   @Override
   public void onProcessState(ProcessId processId, Lifecycle.State to) {
-    if (to == Lifecycle.State.STOPPED) {
-      onProcessStop(processId);
+    switch (to) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Complete cases by adding the missing enum constants or add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this "switch" statement by "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1301)